interpreter/text/parser.mly

```diff
@@ -85,6 +85,17 @@ let nat32 s at =
 let name s at =
   try Utf8.decode s with Utf8.Utf8 -> error at "malformed UTF-8 encoding"
 
+let shuffle_literal ( ss : string Source.phrase list) at : int list =
+  if not (List.length ss = 16) then
+    error at "invalid lane length";
+  let to_index s at =
+    let n = nat s at in
+    if n > 31 then
```

Comment:
This should be like extract/replace and allow up to 0xff, right? In general, can this share behavior from those instructions?

Reply:
Sorry I made this draft because I ran to lunch before rebasing this properly on another PR, I'll fix them up, it should share the same logic as extract replace like you said.

Reply:
Rebased, and fixed.